src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -100,14 +100,17 @@ class AdjustClipBorder final : public WaveClipAdjustBorderHandle::AdjustPolicy
    int mDragStartX{ };
    std::pair<double, double> mRange;
    const bool mAdjustingLeftBorder;
+   const bool mIsStretchMode;
    AdjustHandler mAdjustHandler;
 
    std::unique_ptr<SnapManager> mSnapManager;
    SnapResults mSnap;
 
    void TrimTo(double t) const
    {
-      t = std::clamp(t, mRange.first, mRange.second);
+      if (!mIsStretchMode)
+         // Stretching is allowed outwards, not trimming.
```

Comment:
Reviewing the first commit.  Rather than removing all bounds checks on t, I think you should just change the initialization of `mRange`.  See how there are checks on the neighboring clips, to limit the drag so that clips do not overlap.

Are you sure that stretching will still also avoid clip overlap?  Also will it avoid dragging the left end of the clip beyond the right end?  (Though I suppose negative stretch on a clip could make sense in theory)

Reply:
Indeed that's a better approach. No, stretching could have led to overlapping clips. I pushed a fix commit on top of this branch.